test/test_job.js

```diff
@@ -75,6 +75,41 @@ describe('Job', function(){
         });
     });
 
+    it('fails to remove a locked job', function() {
+      return Job.create(queue, 1, {foo: 'bar'}).then(function(job) {
+        return job.takeLock().then(function(lock) {
+          expect(lock).to.be(true);
+        }).then(function() {
+          return job.remove();
+        }).then(function() {
+          throw new Error('Should not be able to remove a locked job');
+        }).catch(function(err) {
+          expect(err.message).to.equal('Could not get lock for job: ' + job.jobId + '. Cannot remove job.');
+        });
+      });
+    });
+
+    it('removes any job from active set', function() {
+      return queue.add({ foo: 'bar' }).then(function(job) {
+        // Simulate a job in active state but not locked
```

Comment:
@bradvogel @andrewtamura This test should simulate the scenario you guys encountered last week.